app/routes/g_.$groupId.e.new.tsx

```diff
@@ -0,0 +1,77 @@
+import type { FormEvent } from 'react';
+import type { ActionFunctionArgs } from '@remix-run/node';
+import { redirect } from '@remix-run/node';
+import type { FormMethod } from '@remix-run/react';
+import { Form, useParams, useSubmit } from '@remix-run/react';
+
+import { db } from '~/modules/database/db.server';
+import { Card } from '~/components/ui/containers';
+import { Input, TextArea } from '~/components/ui/forms';
+import { Button } from '~/components/ui/button';
+import { H1 } from '~/components/ui/headers';
+
+export async function action({ request }: ActionFunctionArgs) {
+  const form = await request.formData();
+  const name = form.get('name') as string | null;
+  const date = form.get('date') as Date | null;
+  const description = form.get('description') as string | null;
+  const groupId = form.get('groupId') as string | null;
+
+  if (!groupId) {
+    throw Error('Group ID not Provided.');
+  }
+
+  const eventData = {
+    name: name || '',
+    date: date ? new Date(date) : new Date(), // TODO: make DB supports null too?
+    description: description || '',
+    groupId: groupId,
+  };
+
+  await db.event.create({ data: eventData });
+
+  return redirect(`/groups/${groupId}`);
```

Comment:
We could potentially navigate to the new events details page here in the future!